### PR TITLE
ci: exit pre-release changesets when comment created [SFUI2-1184]

### PR DESCRIPTION
**Pull request by Szymon-dziewonski**
# Related issue

https://vsf.atlassian.net/browse/SFUI2-1184

# Scope of work

Github action that is triggered on comment creation to exit pre-release mode for changset, its run when:
- when comment is CREATED in PULL REQUEST, can be issue etc
- when comment value is `!prod-ready`
- when branch name starts with `v2-release/**`

NOTE: this PR will be merged directly into `v2` branch !!

# Screenshots of visual changes

# Checklist

- [x] Self code-reviewed
- [ ] Changes documented
- [ ] Semantic HTML
- [ ] SSR-friendly
- [ ] Caching friendly
- [ ] a11y for WCAG 2.0 AA
- [ ] examples created
- [ ] blocks created
- [ ] cypress tests created

### CODE REVIEW

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      - name: set up env branch name var
+        run: echo "BRANCH_NAME=$(echo $(git branch --show-current))" >> $GITHUB_ENV
+      - name: Exit pre release mode for changesets
+        if: ${{ startsWith( env.BRANCH_NAME, 'v2-release' ) }}
+        run: yarn changeset pre exit
+      - name: Create PR with changelog
+        id: changesets
+        uses: changesets/action@v1
+        with:
+          commit: '[ci] - release'
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:44]:
```suggestion
          commit: '[ci] - release [no ci]'
```

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+name: Exit pre-release
+
+on:
+  workflow_dispatch:
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:4]:
```suggestion
```

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  issue_comment:
+    types: [created]
+
+defaults:
+  run:
+    shell: bash
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:10]:
```suggestion
```

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      - name: Install Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version-file: ".node-version"
+          cache: "yarn"
+      - name: Install dependencies
+        run: yarn --immutable
+      - name: resolve pr refs
+        id: refs
+        uses: eficode/resolve-pr-refs@main
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:29]: let's use some concrete version, e.g. commit hash

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ steps.refs.outputs.head_ref }}
+      - name: set up env branch name var
+        run: echo "BRANCH_NAME=$(echo $(git branch --show-current))" >> $GITHUB_ENV
+      - name: Exit pre release mode for changesets
+        if: ${{ startsWith( env.BRANCH_NAME, 'v2-release' ) }}
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:38]: Maybe we could try to write a condition like [this](https://github.com/vuestorefront/storefront-ui/pull/2797/files#diff-4d183e7fe30b407ed91c82a3f4d3d3d51eef369bd06fe0a180bd04cf74f4e50bR14)?

Comment (Szymon-dziewonski) [reply .github/workflows/exit-pre-mode-on-comment.yml:38]: `steps.refs.outputs.head_ref` returns SHA of latest commit pushed into branch. And I can't get branch name with any `github` event

File: .github/workflows/exit-pre-mode-on-comment.yml (hunk 4 above)
Comment (Szymon-dziewonski) [reply .github/workflows/exit-pre-mode-on-comment.yml]: https://github.com/vuestorefront/storefront-ui/pull/2794/commits/3500821d52c389ccdc415ca93359d9bb0ee394dd

File: .github/workflows/exit-pre-mode-on-comment.yml (hunk 3 above)
Comment (Szymon-dziewonski) [reply .github/workflows/exit-pre-mode-on-comment.yml]: https://github.com/vuestorefront/storefront-ui/pull/2794/commits/3500821d52c389ccdc415ca93359d9bb0ee394dd

File: .github/workflows/exit-pre-mode-on-comment.yml (hunk 1 above)
Comment (Szymon-dziewonski) [reply .github/workflows/exit-pre-mode-on-comment.yml]: https://github.com/vuestorefront/storefront-ui/pull/2794/commits/3500821d52c389ccdc415ca93359d9bb0ee394dd

File: .github/workflows/exit-pre-mode-on-comment.yml [hunk 6]
```diff
@@ -0,0 +1,44 @@
+name: Exit pre-release changesets
+on:
+  issue_comment:
+    types:
+      - created
+
+jobs:
+  exit-pre-release:
+    name: Exit changesets pre-release mode when !prod-ready comment created
+    if: ${{ github.event.comment.body == '!prod-ready' && github.event.issue.pull_request }}
```
Comment (FRSgit) [inline .github/workflows/exit-pre-mode-on-comment.yml:10]: Can we do it in the similar fashion as here?
```suggestion
    if: ${{ contains(github.event.comment.body, '!prod-ready') && github.event.issue.pull_request }}
```

Comment (FRSgit) [reply .github/workflows/exit-pre-mode-on-comment.yml:10]: At least it will be unified

**Review by FRSgit — APPROVED** [review]

**Pull request merged**